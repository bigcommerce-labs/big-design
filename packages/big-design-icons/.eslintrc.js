module.exports = require('@bigcommerce/configs/eslint');
