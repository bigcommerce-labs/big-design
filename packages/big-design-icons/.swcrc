{
  "jsc": {
    "externalHelpers": true,
    "parser": {
      "syntax": "typescript",
      "tsx": true
    }
  },
  "module": {
    "type": "es6"
  },
  "env": {}
}
