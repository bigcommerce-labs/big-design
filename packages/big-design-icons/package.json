{
  "name": "@bigcommerce/big-design-icons",
  "version": "0.26.1",
  "sideEffects": false,
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "typings": "dist/index.d.ts",
  "license": "(MIT AND CC-BY-3.0)",
  "author": "BigCommerce Inc",
  "repository": {
    "type": "git",
    "url": "https://github.com/bigcommerce/big-design.git",
    "directory": "packages/big-design-icons"
  },
  "scripts": {
    "build": "pnpm run build:cjs && pnpm run build:es && pnpm run build:dt",
    "build:cjs": "NODE_ENV=production BABEL_ENV=cjs babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/cjs",
    "build:es": "NODE_ENV=production BABEL_ENV=es babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/es",
    "build:dt": "tsc --emitDeclarationOnly",
    "build:icons": "node scripts/build.js",
    "build:flags": "node scripts/build-flags.js",
    "clean": "rimraf dist && rimraf .tsbuildinfo",
    "download": "node scripts/downloader.js",
    "prepublishOnly": "pnpm run clean && pnpm run typecheck && pnpm run build",
    "typecheck": "tsc --noEmit"
  },
  "files": [
    "dist",
    "flags",
    "svgs"
  ],
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@babel/runtime": "^7.15.4",
    "@bigcommerce/big-design-theme": "workspace:^"
  },
  "peerDependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "styled-components": "^5.3.5"
  },
  "devDependencies": {
    "@babel/cli": "^7.15.4",
    "@babel/core": "^7.15.5",
    "@babel/plugin-transform-runtime": "^7.15.0",
    "@babel/preset-env": "^7.15.6",
    "@babel/preset-react": "^7.12.1",
    "@babel/preset-typescript": "^7.15.0",
    "@bigcommerce/configs": "^0.20.1",
    "@svgr/core": "^8.1.0",
    "@svgr/plugin-jsx": "^8.1.0",
    "@svgr/plugin-prettier": "^8.1.0",
    "@svgr/plugin-svgo": "^8.1.0",
    "@types/react": "^18.2.73",
    "@types/react-dom": "^18.2.23",
    "@types/styled-components": "^5.1.26",
    "babel-plugin-styled-components": "^2.0.7",
    "camelcase": "^6.3.0",
    "flag-icons": "^7.1.0",
    "fs-extra": "^11.1.0",
    "glob": "^10.3.12",
    "inquirer": "^8.2.4",
    "inquirer-autocomplete-prompt": "^2.0.0",
    "prettier": "^2.4.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-is": "^18.2.0",
    "rimraf": "^5.0.0",
    "styled-components": "^5.3.5",
    "tiny-async-pool": "^2.0.1",
    "typescript": "^5.4.4",
    "typescript-styled-plugin": "^0.18.2"
  }
}
