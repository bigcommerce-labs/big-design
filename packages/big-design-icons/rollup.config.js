import { tmpdir } from 'os';
import { DEFAULT_EXTENSIONS } from '@babel/core';
import babel from 'rollup-plugin-babel';
import commonjs from 'rollup-plugin-commonjs';
import filesize from 'rollup-plugin-filesize';
import resolve from 'rollup-plugin-node-resolve';
import replace from 'rollup-plugin-replace';
import sourceMaps from 'rollup-plugin-sourcemaps';
import typescript from 'rollup-plugin-typescript2';

import pkg from './package.json';

const nodeEnv = process.env.NODE_ENV || 'production';
const externals = [...Object.keys(pkg.dependencies || {}), ...Object.keys(pkg.peerDependencies || {})];

export default {
  input: 'src/index.ts',
  output: [{ file: pkg.main, format: 'cjs', sourcemap: true }, { file: pkg.module, format: 'es', sourcemap: true }],
  external: makeExternalPredicate(externals),
  watch: {
    include: 'src/**',
  },
  plugins: [
    replace({
      'process.env.NODE_ENV': JSON.stringify(nodeEnv),
    }),
    typescript({
      cacheRoot: tmpdir(),
      check: nodeEnv === 'production',
      typescript: require('typescript'),
      useTsconfigDeclarationDir: true,
    }),
    babel({
      extensions: [...DEFAULT_EXTENSIONS, 'ts', 'tsx'],
      exclude: 'node_modules/**',
    }),
    commonjs(),
    resolve(),
    sourceMaps(),
    filesize(),
  ],
};

function makeExternalPredicate(externalsArr) {
  if (externalsArr.length === 0) {
    return () => false;
  }

  const externalPattern = new RegExp(`^(${externalsArr.join('|')})($|/)`);

  return id => externalPattern.test(id);
}
