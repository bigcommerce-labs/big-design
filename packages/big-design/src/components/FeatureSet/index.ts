export { FeatureSet, type FeatureSetProps } from './FeatureSet';
