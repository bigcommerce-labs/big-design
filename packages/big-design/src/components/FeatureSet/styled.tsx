import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { withMargins } from '../../helpers';

export const StyledUl = styled.ul.attrs({ theme: defaultTheme })`
  ${({ theme }) => theme.helpers.listReset}

  ${withMargins()};

  display: inline-flex;
  flex-wrap: wrap;
  gap: ${({ theme }) => theme.spacing.small};
`;
