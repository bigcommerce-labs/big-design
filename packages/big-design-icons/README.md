# BigDesign Icons

BigDesign Icons as React components.

### Documentation

You can find documentation, list of icons, and examples on our [docs page](https://bigcommerce.github.io/big-design/?path=/story/icons--overview).

### Quick start guide

Add BigDesign Icons and styled-components to your project using `npm`:

```
npm install @bigcommerce/big-design-icons styled-components
```

or with `yarn`:

```
yarn add @bigcommerce/big-design-icons styled-components
```

Import any icon component and use it anywhere in your app.

```jsx
import { StoreIcon } from '@bigcommerce/big-design-icons';

...

<App>
  <StoreIcon />
</App>
```
