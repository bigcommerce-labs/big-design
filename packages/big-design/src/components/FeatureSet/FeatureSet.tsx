import React, { ComponentPropsWithoutRef, memo } from 'react';

import { MarginProps } from '../../helpers';
import { FeatureTag, FeatureTagProps } from '../FeatureTag';

import { StyledFeatureSet } from './styled';

export interface FeatureSetProps extends ComponentPropsWithoutRef<'ul'>, MarginProps {
  features: FeatureTagProps[];
}

export const FeatureSet: React.FC<FeatureSetProps> = memo(({ features, ...props }) => {
  return features && features.length > 0 ? (
    <StyledFeatureSet {...props}>
      {features.map((feature, index) => (
        <li key={index}>
          <FeatureTag {...feature} />
        </li>
      ))}
    </StyledFeatureSet>
  ) : null;
});

FeatureSet.displayName = 'FeatureSet';
