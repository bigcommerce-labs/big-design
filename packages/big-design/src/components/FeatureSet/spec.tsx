import { AutoAwesomeIcon } from '@bigcommerce/big-design-icons';
import { render } from '@testing-library/react';
import React from 'react';

import 'jest-styled-components';

import { FeatureSet } from './FeatureSet';

test('render feature set', () => {
  const { container } = render(
    <FeatureSet
      features={[
        { icon: <AutoAwesomeIcon />, label: 'Feature 1' },
        { icon: <AutoAwesomeIcon />, label: 'Feature 2' },
      ]}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test("doesn't render if feature prop is absent", () => {
  // @ts-expect-error ignoring since label prop is required
  const { container } = render(<FeatureSet />);

  expect(container.firstChild).toBeNull();
});

test("doesn't render if feature prop array length is empty", () => {
  const { container } = render(<FeatureSet features={[]} />);

  expect(container.firstChild).toBeNull();
});
