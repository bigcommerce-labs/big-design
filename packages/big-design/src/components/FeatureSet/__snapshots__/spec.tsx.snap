// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render feature set 1`] = `
.c2 {
  vertical-align: middle;
  height: 1.5rem;
  width: 1.5rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  background-color: #ECEEF5;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #5E637A;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  font-size: 0.875rem;
  gap: 0.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.25rem;
  max-width: 100%;
  padding-block: 0.125rem;
  padding-inline: 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 > svg {
  height: 1rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 1rem;
}

.c3 {
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  width: 100%;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c0 > li {
  display: block;
  margin-right: 0.75rem;
  margin-bottom: 0.75rem;
  overflow: hidden;
}

<ul
  class="c0"
>
  <li>
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="m19.46 8 .79-1.75L22 5.46a.5.5 0 0 0 0-.91l-1.75-.79L19.46 2a.5.5 0 0 0-.91 0l-.79 1.75-1.76.79a.5.5 0 0 0 0 .91l1.75.79.79 1.76c.18.39.74.39.92 0M11.5 9.5 9.91 6c-.35-.78-1.47-.78-1.82 0L6.5 9.5 3 11.09c-.78.36-.78 1.47 0 1.82l3.5 1.59L8.09 18c.36.78 1.47.78 1.82 0l1.59-3.5 3.5-1.59c.78-.36.78-1.47 0-1.82zm7.04 6.5-.79 1.75-1.75.79a.5.5 0 0 0 0 .91l1.75.79.79 1.76a.5.5 0 0 0 .91 0l.79-1.75 1.76-.79a.5.5 0 0 0 0-.91l-1.75-.79-.79-1.76a.508.508 0 0 0-.92 0"
        />
      </svg>
      <span
        class="c3"
      >
        Feature 1
      </span>
    </span>
  </li>
  <li>
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="m19.46 8 .79-1.75L22 5.46a.5.5 0 0 0 0-.91l-1.75-.79L19.46 2a.5.5 0 0 0-.91 0l-.79 1.75-1.76.79a.5.5 0 0 0 0 .91l1.75.79.79 1.76c.18.39.74.39.92 0M11.5 9.5 9.91 6c-.35-.78-1.47-.78-1.82 0L6.5 9.5 3 11.09c-.78.36-.78 1.47 0 1.82l3.5 1.59L8.09 18c.36.78 1.47.78 1.82 0l1.59-3.5 3.5-1.59c.78-.36.78-1.47 0-1.82zm7.04 6.5-.79 1.75-1.75.79a.5.5 0 0 0 0 .91l1.75.79.79 1.76a.5.5 0 0 0 .91 0l.79-1.75 1.76-.79a.5.5 0 0 0 0-.91l-1.75-.79-.79-1.76a.508.508 0 0 0-.92 0"
        />
      </svg>
      <span
        class="c3"
      >
        Feature 2
      </span>
    </span>
  </li>
</ul>
`;
